src/aws-cpp-sdk-core/source/utils/logging/CRTLogging.cpp

```diff
@@ -128,39 +97,33 @@ static struct aws_logger_vtable s_aws_logger_redirect_vtable = {
 /**
  * Installs SDK wrapper over CRT logger.
  * This wrapper will redirect all CRT logger calls to the set SDK "CRTLogSystem"
- *   (or back to an original CRT logger if SDK API is terminated).
  * This method is not thread-safe (as the most other global Init/Shutdown APIs of the SDK).
  */
 void SetUpCrtLogsRedirection()
 {
-    if (!s_CRTLogsRedirectionIsSet)
-    {
-        static std::once_flag flag;
-        std::call_once(flag, [&]()
-        {
-            s_sdkCrtLogger.vtable = &s_aws_logger_redirect_vtable;
-            s_sdkCrtLogger.allocator = Aws::get_aws_allocator();
-            s_sdkCrtLogger.p_impl = &s_sdkCrtLogger;
-
-            auto pPrevLogger = aws_logger_get();
-            if (pPrevLogger) {
-                s_previousCrtLogger = *pPrevLogger;
-            }
-            aws_logger_set(&s_sdkCrtLogger);
-            s_CRTLogsRedirectionIsSet = true;
-        });
-    }
+    s_sdkCrtLogger.vtable = &s_aws_logger_redirect_vtable;
+    s_sdkCrtLogger.allocator = Aws::get_aws_allocator();
+    s_sdkCrtLogger.p_impl = &s_sdkCrtLogger;
+
+    aws_logger_set(&s_sdkCrtLogger);
 }
 
 void InitializeCRTLogging(const std::shared_ptr<CRTLogSystemInterface>& inputCrtLogSystem) {
-    Aws::Utils::Threading::WriterLockGuard g(CRTLogSystemLock);
     SetUpCrtLogsRedirection();
     CRTLogSystem = inputCrtLogSystem;
 }
 
 void ShutdownCRTLogging() {
-    Aws::Utils::Threading::WriterLockGuard g(CRTLogSystemLock);
-    CRTLogSystem.reset();
+    if (aws_logger_get() == &s_sdkCrtLogger)
+    {
+        aws_logger_set(nullptr);
+    }
+    // GetLogSystem returns a raw pointer
+    // so this is a hack to let all other threads finish their log statement after getting a LogSystem pointer
+    // otherwise we would need to perform ref-counting on each logging statement
+    auto tmpLogger = std::move(CRTLogSystem);
```

Comment:
This approach would not be necessary if the logger's lifetime always exceeded the CRT.  We should be able to shutdown the CRT _and then_ call aws_logger_set(nullptr) and have no need for time-based quiesces which are ultimately a flawed approach (someone will always exceed the sleep time).  We would then get all the log messages from CRT shutdown.  We're shutting down CRT logging before CRT is stopped, and hooking CRT logging in after CRT is started.  There's a more fundamental problem here.